File: Stores/EMOTpackage/private/plotResults.m
function [] = plotResults(x,y,res,pars,subj,names,time)


%% Computing useful graphic parameters
xC2 = linspace(angleX(pars.d(1),pars.d(2)), angleX(pars.a(1),pars.a(2)),100);
xC1 = linspace(angleX(-pars.b(1),pars.b(2)), angleX(-pars.c(1),pars.c(2)),100);
rescaled_xC2 = xC2' + repmat(360-180,length(xC2),1);
rescaled_xC2(rescaled_xC2>=360) = rescaled_xC2(rescaled_xC2>=360) - 360;
rescaled_xC1 = xC1' + repmat(360-180,length(xC1),1);
rescaled_xC1(rescaled_xC1>=360) = rescaled_xC1(rescaled_xC1>=360) - 360;
pols_xC1_rad = degtorad(rescaled_xC1); pols_xC2_rad = degtorad(rescaled_xC2);
radians_lbls = [pols_xC1_rad(1) pols_xC2_rad(end)];
radians_lbls2 = [pols_xC1_rad pols_xC2_rad];
k = find(res.xHist==median(res.xHist));
xHist_zeta1 = res.xHist(1:k-1); xHist_zeta2 = res.xHist(k+1:end);



%% Create figure
close all
hFig = figure(1);
set(hFig, 'Position', [20 100 1200 600], 'Color',[0.8 0.8 0.8],'Name','EMOT analysis')

%% Empirical mouse-movements
subplot(3,7,[1 10]);hold on
rectangle('Position',[pars.c(1), pars.c(2), abs(pars.c(1)-pars.b(1)),abs(pars.d(2)-pars.c(2))],'LineWidth',1,'LineStyle','--','FaceColor',[1,1,1])
rectangle('Position',[-pars.b(1), pars.b(2), abs(pars.c(1)-pars.b(1)),abs(pars.d(2)-pars.c(2))],'LineWidth',1,'LineStyle','--','FaceColor',[1,1,1])
text(pars.b(1)-0.1,pars.d(2)+0.05,'C2','Color','black','FontSize',14,'FontName','Arial')
text(-pars.b(1),pars.d(2)+0.05,'C1','Color','black','FontSize',14,'FontName','Arial')
text(pars.c(1)+0.05,pars.d(2)/1.15,upper(names(1)),'Color','black','FontSize',17,'FontName','Arial')
text(-pars.b(1)+0.05,pars.d(2)/1.15,upper(names(2)),'Color','black','FontSize',17,'FontName','Arial')
text(x(1)-0.17,y(1)-0.17,upper(names(3)),'Color','black','FontSize',17,'FontName','Arial')
xlim([-pars.b(1)-0.3 pars.b(1)+0.3]);
ylim([0 pars.d(2)+0.2])
scatter(x,y,'b')
plot(linspace(-2,2,200),res.thr)
title(['\fontsize{18} Participant ID: '    num2str(subj)],'fontweight','bold','FontName','Arial')

%% Histogram H_theta and H_theta0
subplot(3,7,[4 13]);hold on
bar(res.xHist,(res.H_theta/sum(res.H_theta)),1,'facecolor',[0.6 0.6 0.6])
bar(res.xHist,(res.H_theta0/sum(res.H_theta0)+1e-19),1,'facecolor',[0.9 0.9 0.9])
xlim([min(res.xHist)-0.20 max(res.xHist)+0.20])
plot(radians_lbls2(:,2),0,'-ks','markers',2,'Color', [0.05 0.05 0.05]);plot(radians_lbls2(:,1),0,'-ks','markers',2,'Color', [0.05 0.05 0.05])
text(mean(radians_lbls2(:,1)),-0.028,'C1','Color','black','FontSize',14,'FontName','Arial')
text(mean(radians_lbls2(:,2)),-0.028,'C2','Color','black','FontSize',14,'FontName','Arial')
legend('\fontsize{15} H_\theta','\fontsize{15} H_{\theta0}','Location','northoutside','Orientation','horizontal')
legend('boxoff')

%% Histograms CSI
subplot(3,7,[17.5 18.5]);hold on
bar(res.xHist,res.tau,1,'facecolor',[0.9 0.9 0.9])
legend('\fontsize{15}\tau','Location','northoutside','Orientation','vertical')
legend('boxoff')
xlim([min(res.xHist)-0.20 max(res.xHist)+0.20])

%% Histograms ZETA1 e ZETA2
subplot(3,7,[19.5 20.5]);hold on
bar(xHist_zeta1,res.u1,1,'facecolor',[0.9 0.9 0.9])
bar(xHist_zeta2,res.u2,1,'facecolor',[0.6 0.6 0.6])
legend('\fontsize{14}\upsilon_{ 1}','\fontsize{14}\upsilon_{ 2}','Location','northoutside','Orientation','horizontal')
legend('boxoff')
xlim([min(xHist_zeta1)-0.20 max(xHist_zeta2)+0.20])

%% Text of results
fontText=16;
subplot(3,7,[15]);hold on; axis('off'); 
text(0,0.8,['\psi = ' num2str(res.psi)],'Color','black','FontSize',fontText,'FontName','Arial')
text(1.4,0.8,['\xi = ' num2str(res.csi)],'Color','black','FontSize',fontText,'FontName','Arial')
text(0,0.6,['\zeta = ' num2str(res.zeta)],'Color','black','FontSize',fontText,'FontName','Arial')
text(1.4,0.55,['\zeta_{ 1} = ' num2str(res.zeta1)],'Color','black','FontSize',fontText,'FontName','Arial')
text(0,0.35,['\zeta_{ 2} = ' num2str(res.zeta2)],'Color','black','FontSize',fontText,'FontName','Arial')
text(1.4,0.4,['t = ' num2str(time(1)) ' (' num2str(time(2)) ') ms'],'Color','black','FontSize',fontText,'FontName','Arial')




end

File: script/EFIT/plot_Results.py
import numpy as np
import matplotlib.pyplot as plt

def plot_Results(x, y, res, pars, subj, names, time):

    # Computing useful graphic parameters
    angleX = np.vectorize(lambda x1, x2: np.arctan2(x2, x1) * 180 / np.pi)
    xC2 = np.linspace(angleX(pars['d'][0], pars['d'][1]), angleX(pars['a'][0], pars['a'][1]), 100)
    xC1 = np.linspace(angleX(-pars['b'][0], pars['b'][1]), angleX(-pars['c'][0], pars['c'][1]), 100)
    rescaled_xC2 = np.mod(xC2 + (360 - 180), 360)
    rescaled_xC1 = np.mod(xC1 + (360 - 180), 360)
    pols_xC1_rad = np.deg2rad(rescaled_xC1)
    pols_xC2_rad = np.deg2rad(rescaled_xC2)
    radians_lbls = [pols_xC1_rad[0], pols_xC2_rad[-1]]
    radians_lbls2 = np.concatenate((pols_xC1_rad, pols_xC2_rad))
    k = np.where(res['xHist'] == np.median(res['xHist']))[0][0]
    xHist_zeta1 = res['xHist'][:k]
    xHist_zeta2 = res['xHist'][k+1:]

    # Create figure
    plt.close('all')
    fig = plt.figure(figsize=(17, 8.5), dpi=80)
    fig.canvas.set_window_title('EMOT analysis')

    # Empirical mouse-movements
    ax1 = plt.subplot(3, 7, (1, 10))
    ax1.add_patch(plt.Rectangle((pars['c'][0], pars['c'][1]), abs(pars['c'][0] - pars['b'][0]), abs(pars['d'][1] - pars['c'][1]), linewidth=1, linestyle='--', facecolor='white'))
    ax1.add_patch(plt.Rectangle((-pars['b'][0], pars['b'][1]), abs(pars['c'][0] - pars['b'][0]), abs(pars['d'][1] - pars['c'][1]), linewidth=1, linestyle='--', facecolor='white'))
    ax1.text(pars['b'][0] - 0.1, pars['d'][1] + 0.05, 'C2', color='black', fontsize=14, fontname='Arial')
    ax1.text(-pars['b'][0], pars['d'][1] + 0.05, 'C1', color='black', fontsize=14, fontname='Arial')
    ax1.text(pars['c'][0] + 0.05, pars['d'][1] / 1.15, names[0].upper(), color='black', fontsize=17, fontname='Arial')
    ax1.text(-pars['b'][0] + 0.05, pars['d'][1] / 1.15, names[1].upper(), color='black', fontsize=17, fontname='Arial')
    ax1.text(x[0] - 0.17, y[0] - 0.17, names[2].upper(), color='black', fontsize=17, fontname='Arial')
    ax1.set_xlim([-pars['b'][0] - 0.3, pars['b'][0] + 0.3])
    ax1.set_ylim([0, pars['d'][1] + 0.2])
    ax1.scatter(x, y, color='b')
    ax1.plot(np.linspace(-2, 2, 200), res['thr'])
    ax1.set_title(f'Participant ID: {subj}', fontsize=18, fontweight='bold', fontname='Arial')

    # Histogram H_theta and H_theta0
    ax2 = plt.subplot(3, 7, (4, 13))
    ax2.bar(res['xHist'], res['H_theta'] / np.sum(res['H_theta']), color=[0.6, 0.6, 0.6])
    ax2.bar(res['xHist'], (res['H_theta0'] / np.sum(res['H_theta0']) + 1e-19), color=[0.9, 0.9, 0.9])
    ax2.set_xlim([np.min(res['xHist']) - 0.20, np.max(res['xHist']) + 0.20])
    ax2.plot(radians_lbls2, np.zeros_like(radians_lbls2), '-ks', markersize=2, color=[0.05, 0.05, 0.05])
    ax2.text(np.mean(radians_lbls2[:len(radians_lbls2)//2]), -0.028, 'C1', color='black', fontsize=14, fontname='Arial')
    ax2.text(np.mean(radians_lbls2[len(radians_lbls2)//2:]), -0.028, 'C2', color='black', fontsize=14, fontname='Arial')
    ax2.legend(['H_theta', 'H_theta0'], fontsize=15, loc='upper center', bbox_to_anchor=(0.5, 1.1), ncol=2)
    ax2.get_legend().set_title(None)

    # Histograms CSI
    ax3 = plt.subplot(3, 7, (25, 26))
    ax3.bar(res['xHist'], res['tau'], color=[0.9, 0.9, 0.9])
    ax3.legend(['tau'], fontsize=15, loc='upper center', bbox_to_anchor=(0.5, 1.2), ncol=1)
    ax3.get_legend().set_title(None)
    ax3.set_xlim([np.min(res['xHist']) - 0.20, np.max(res['xHist']) + 0.20])

    # Histograms ZETA1 e ZETA2
    ax4 = plt.subplot(3, 7, (27, 28))
    ax4.bar(xHist_zeta1, res['u1'], color=[0.9, 0.9, 0.9])
    ax4.bar(xHist_zeta2, res['u2'], color=[0.6, 0.6, 0.6])
    ax4.legend(['upsilon_1', 'upsilon_2'], fontsize=14, loc='upper center', bbox_to_anchor=(0.5, 1.1), ncol=2)
    ax4.get_legend().set_title(None)
    ax4.set_xlim([np.min(xHist_zeta1) - 0.20, np.max(xHist_zeta2) + 0.20])

    # Text of results
    fontText = 16
    ax5 = plt.subplot(3, 7, 15)
    ax5.text(0, 0.8, f'psi = {res["psi"]}', color='black', fontsize=fontText, fontname='Arial')
    ax5.text(1.4, 0.8, f'xi = {res["csi"]}', color='black', fontsize=fontText, fontname='Arial')
    ax5.text(0, 0.6, f'zeta = {res["zeta"]}', color='black', fontsize=fontText, fontname='Arial')
    ax5.text(1.4, 0.55, f'zeta_1 = {res["zeta1"]}', color='black', fontsize=fontText, fontname='Arial')
    ax5.text(0, 0.35, f'zeta_2 = {res["zeta2"]}', color='black', fontsize=fontText, fontname='Arial')
    ax5.text(1.4, 0.4, f't = {time[0]} ({time[1]}) ms', color='black', fontsize=fontText, fontname='Arial')
    ax5.axis('off')

    plt.tight_layout()
    plt.show()